src/Polly.Core/Builder/ResilienceStrategyPipeline.cs

```diff
@@ -1,55 +1,46 @@
 using System;
+using static System.Net.Mime.MediaTypeNames;
 
 namespace Polly.Builder;
 
+#pragma warning disable S2302 // "nameof" should be used
+
 /// <summary>
 /// A pipeline of strategies.
 /// </summary>
-internal sealed class ResilienceStrategyPipeline : DelegatingResilienceStrategy
+internal sealed class ResilienceStrategyPipeline : ResilienceStrategy
 {
     private readonly ResilienceStrategy _pipeline;
 
-    public static ResilienceStrategyPipeline CreatePipelineAndFreezeStrategies(IReadOnlyList<ResilienceStrategy> strategies)
+    public static ResilienceStrategyPipeline CreatePipeline(IReadOnlyList<ResilienceStrategy> strategies)
     {
         Guard.NotNull(strategies);
 
         if (strategies.Count < 2)
         {
-#pragma warning disable S2302 // "nameof" should be used
             throw new InvalidOperationException("The resilience pipeline must contain at least two resilience strategies.");
-#pragma warning restore S2302 // "nameof" should be used
         }
 
         if (strategies.Distinct().Count() != strategies.Count)
         {
-#pragma warning disable S2302 // "nameof" should be used
             throw new InvalidOperationException("The resilience pipeline must contain unique resilience strategies.");
-#pragma warning restore S2302 // "nameof" should be used
         }
 
-        var delegatingStrategies = strategies.Select(strategy =>
-        {
-            if (strategy is DelegatingResilienceStrategy delegatingStrategy)
-            {
-                return delegatingStrategy;
-            }
-            else
-            {
-                return new DelegatingStrategyWrapper(strategy);
-            }
-        }).ToList();
+        // convert all strategies to delegating ones (except the last one as it's not required)
+        var delegatingStrategies = strategies
+            .Take(strategies.Count - 1)
+            .Select(strategy => new DelegatingResilienceStrategy(strategy))
+            .ToList();
 
+        // link the last one
+        delegatingStrategies.Last().Next = strategies[strategies.Count - 1];
```

Comment:
Can we use indexer over LINQ here?

```suggestion
        delegatingStrategies[^1].Next = strategies[strategies.Count - 1];
```

Reply:
Dropped the ling and on .NET 6+ used the simpler syntax